rfcs/2021-12-01-9460-healthcheck-endpoint-improvements.md

```diff
@@ -0,0 +1,115 @@
+# RFC 9460 - 2021-12-01 - Healthcheck endpoint improvements
+
+Our existing health endpoint is too limited and additional features are required
+to properly interact with load balancers or Kubernetes service discovery.
+
+## Context
+
+- [Additional response for api health endpoint](https://github.com/vectordotdev/vector/issues/9160)
+
+## Scope
+
+### In scope
+
+- Update health endpoint to improve Vector's reliability in production deployments
+- Update default configurations to improve Vector's out-of-the-box reliability
+
+### Out of scope
+
+- Moving the health endpoint out of our existing API
```

Comment:
I'd go further than deeming this out out of scope, and suggest this should necessarily remain coupled for valid technical reasons.

As I [mentioned here](https://github.com/vectordotdev/vector/pull/10297#discussion_r769391071), this was bundled with API code originally because both share a common Warp HTTP server. If we were to move it out of API, that would suggest we'd either need _two_ HTTP servers (complicating the port setup) - or, some other HTTP abstraction that would allow modules such as API and health to separately register routes. I think the latter would be an unnecessary complication.

Reply:
👍 I talked some with @jszwedko last night about conditionally attaching the graphql handler so we could always expose the health endpoint but not the graphql paths, which some users have expressed concerns over having enabled for security reasons.

Reply:
Agreed, `api = false` & `health = true` would be useful.

I still think the coupling makes sense. Like `playground = false`, it'd just be a case of turning on/off certain routes.